extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -698,4 +774,29 @@ export class Bind {
 
     return false;
   }
+
+  /**
+   * Wait for DOM mutation observer callbacks to fire. Returns a promise
+   * that resolves when mutation callbacks have fired.
+   *
+   * @return {Promise}
+   *
+   * @visibleForTesting
+   */
+  waitForAllMutationsForTesting_() {
+    return new Promise(resolve => {
+      (function waitForMutationCallback() {
```

Comment:
Sorry, meant the binding.

```js
const waitForMutationCallback = () => {
  if (this.mutationPromises_.length > 0) {
  } else {
  }
};
```

We could further improve it by extracting [`poll`](https://github.com/ampproject/amphtml/blob/8074543/src/dom.js#L56-L61) out of `waitForChild`.